Zotero/Scenes/Master/Collections/Views/ExpandableCollectionsCollectionViewHandler.swift

```diff
@@ -68,122 +86,55 @@ final class ExpandableCollectionsCollectionViewHandler: NSObject {
+            collectionView.selectItem(at: indexPath, animated: false, scrollPosition: scrollToPosition ? .centeredVertically : [])
         } else if let indexPath = selectedIndexPaths.first {
-            self.collectionView.deselectItem(at: indexPath, animated: false)
+            collectionView.deselectItem(at: indexPath, animated: false)
         }
     }
 
-    // MARK: - Data Source
-
     func update(with tree: CollectionTree, selectedId: CollectionIdentifier, animated: Bool, completion: (() -> Void)? = nil) {
         let newSnapshot = tree.createSnapshot(selectedId: selectedId)
-
-        if self.dataSource.snapshot(for: self.collectionsSection).items.count == newSnapshot.items.count {
-            self.dataSource.apply(newSnapshot, to: self.collectionsSection, animatingDifferences: animated, completion: completion)
-            return
-        }
-
-        // TODO: - iOS bug, applying a section snapshot to section where a new child row is added, parent row doesn't show a collapse button
-        // It works fine if a completely new snapshot is applied, which breaks animations though.
-
-        var snapshot = NSDiffableDataSourceSnapshot<Int, Collection>()
-        snapshot.appendSections([self.collectionsSection])
-        self.dataSource.apply(snapshot, animatingDifferences: animated)
-        self.dataSource.apply(newSnapshot, to: self.collectionsSection, animatingDifferences: animated, completion: completion)
-    }
-
-    private func createContextMenu(for collection: Collection) -> UIMenu? {
-        var actions: [UIAction] = []
-        switch collection.identifier {
-        case .collection(let key):
-            if viewModel.state.library.metadataEditable {
-                let edit = UIAction(title: L10n.edit, image: UIImage(systemName: "pencil")) { [weak viewModel] _ in
-                    viewModel?.process(action: .startEditing(.edit(collection)))
-                }
-                let subcollection = UIAction(title: L10n.Collections.newSubcollection, image: UIImage(systemName: "folder.badge.plus")) { [weak viewModel] _ in
-                    viewModel?.process(action: .startEditing(.addSubcollection(collection)))
-                }
-                actions.append(contentsOf: [edit, subcollection])
-            }
-            
-            let createBibliography = UIAction(title: L10n.Collections.createBibliography, image: UIImage(systemName: "doc.on.doc")) { [weak viewModel] _ in
-                viewModel?.process(action: .loadItemKeysForBibliography(collection))
-            }
-            actions.append(createBibliography)
-
-            if viewModel.state.library.metadataEditable {
-                let delete = UIAction(title: L10n.moveToTrash, image: UIImage(systemName: "trash"), attributes: .destructive) { [weak viewModel] _ in
-                    viewModel?.process(action: .deleteCollection(key))
+        let mainThreadCompletion: (() -> Void)? = completion.flatMap { action in
```

Comment:
This is not needed, `apply` completion should be called in the main queue.

Reply:
Removed in 0b7a0c9b321df800cc80e3ef60cdc4359d810aa7